compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -137,8 +137,14 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
 
             Assert { ref cond, expected, ref msg, target, cleanup } => {
+                let ignored = !M::checked_binop_checks_overflow(self)
```

Comment:
The old name of this machine hook stopped making sense. Can you rename it to `ignored_checkable_overflow_assertions` or something like that?